standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestRemoteHiveMetastoreWithHttpJwt.java

```diff
@@ -0,0 +1,269 @@
+    MetastoreConf.setVar(conf, ConfVars.STORAGE_SCHEMA_READER_IMPL, "no.such.class");
+
+    setupMockServer();
+    MetastoreConf.setBoolVar(conf, ConfVars.EXECUTE_SET_UGI, false);
+    MetastoreConf.setVar(conf, ConfVars.THRIFT_TRANSPORT_MODE, "http");
+    MetastoreConf.setVar(conf, ConfVars.METASTORE_CLIENT_THRIFT_TRANSPORT_MODE, "http");
+
+    MetastoreConf.setVar(conf, ConfVars.METASTORE_CLIENT_AUTH_MODE, "JWT");
+    MetastoreConf.setVar(conf, ConfVars.THRIFT_METASTORE_AUTHENTICATION, "JWT");
+    startMetastoreServer();
+  }
+
+  private static void startMetastoreServer() throws Exception {
+    port = MetaStoreTestUtils.startMetaStoreWithRetry(HadoopThriftAuthBridge.getBridge(),
+        conf);
+    MetastoreConf.setVar(conf, ConfVars.THRIFT_URIS, "thrift://localhost:" + port);
+    System.out.println("Starting MetaStore Server on port " + port);
+  }
+
+  @Before
+  public void initEnvMap() {
+    envMap.clear();
+    envMap.putAll(DEFAULTS);
+  }
+
+  private static void setupMockServer() throws Exception {
+    MOCK_JWKS_SERVER.stubFor(get("/jwks")
+        .willReturn(ok()
+            .withBody(Files.readAllBytes(jwtVerificationJWKSFile.toPath()))));
+    MetastoreConf.setVar(conf, ConfVars.THRIFT_METASTORE_AUTHENTICATION_JWT_JWKS_URL,
+        "http://localhost:" + MOCK_JWKS_SERVER_PORT + "/jwks");
+  }
+
+  @Test
+  public void testValidJWT() throws Exception {
+    String validJwtToken = generateJWT(USER_1, jwtAuthorizedKeyFile.toPath(),
+        TimeUnit.MINUTES.toMillis(5));
+    System.getenv().put("HMS_JWT", validJwtToken);
+    String dbName = ("valid_jwt_" + TEST_DB_NAME_PREFIX + "_" + UUID.randomUUID()).toLowerCase();
+    HiveMetaStoreClient client = new HiveMetaStoreClient(conf);
+    try {
+      Database createdDb = new Database();
+      createdDb.setName(dbName);
+      client.createDatabase(createdDb);
+      Database dbFromServer = client.getDatabase(dbName);
+      assertEquals(dbName, dbFromServer.getName());
+    } finally {
+      try {
+        client.dropDatabase(dbName);
+      } catch (Exception e) {
+        LOG.warn("Failed to drop database: " + dbName + ". Error message: " + e);
+      }
+      try {
+        client.close();
+      } catch (Exception e) {
+        LOG.error("Failed to close metastore client");
+      }
+    }
+  }
+
+  @Test(expected = TTransportException.class)
+  public void testExpiredJWT() throws Exception {
+    String validJwtToken = generateJWT(USER_1, jwtAuthorizedKeyFile.toPath(),
+        TimeUnit.MILLISECONDS.toMillis(2));
+    System.getenv().put("HMS_JWT", validJwtToken);
+    String dbName = ("expired_jwt_" + TEST_DB_NAME_PREFIX + "_" + UUID.randomUUID()).toLowerCase();
+    HiveMetaStoreClient client = new HiveMetaStoreClient(conf);
+    try {
+      Thread.sleep(TimeUnit.MILLISECONDS.toMillis(2));
+      Database createdDb = new Database();
+      createdDb.setName(dbName);
+      client.createDatabase(createdDb);
```

Comment:
maybe we should check the exception which is caused by expired token here

Reply:
Yes, that is what I wanted to check initially. However, it looks like THttpClient class (which the above client use as underlying transport) catches the exception from the server and throws TTransportException instead.